svf-llvm/lib/ObjTypeInference.cpp

```diff
@@ -790,47 +795,56 @@ Set<const Value *> &ObjTypeInference::bwFindAllocOrClsNameSources(const Value *s
                 }
             }
         }
+
+        // If updating is allowed; store the gathered sources as sources for the current value in the cache
         if (canUpdate)
         {
             _valueToAllocOrClsNameSources[curValue] = sources;
         }
     }
+
     return _valueToAllocOrClsNameSources[startValue];
 }
 
-Set<const Function *> &ObjTypeInference::fwFindClsNameSources(const CallBase *alloc)
+Set<const Value *> &ObjTypeInference::fwFindClsNameSources(const Value *startValue)
 {
+    assert(startValue && "startValue was null?");
+
     // consult cache
-    auto tIt = _allocToClsNameSources.find(alloc);
-    if (tIt != _allocToClsNameSources.end())
+    auto tIt = _objToClsNameSources.find(startValue);
+    if (tIt != _objToClsNameSources.end())
     {
         return tIt->second;
     }
 
-    Set<const Function *> clsSources;
-    // for heap allocation, we forward find class name sources
-    auto inferViaCppCall = [&clsSources](const CallBase *callBase)
+    Set<const Value *> sources;
```

Comment:
using `Set<const CallBase*>` is more precise?

Reply:
Fair enough; I initially had Function objects or CallBases in this set, but now this makes more sense.